DatadogRUM/Sources/RUMConfiguration.swift

```diff
@@ -7,156 +7,341 @@
+            ///
+            /// This closure gets called for each network request intercepted by RUM. Use it to return additional
+            /// attributes for RUM resource based on the provided request, response, data, and error.
+            /// Keep the implementation fast and do not make any assumptions on the thread used to run it.
+            ///
+            /// Default: `nil`.
+            public var resourceAttributesProvider: RUMResourceAttributesProvider?
+
+            /// Private init to avoid `invalid redeclaration of synthesized memberwise init(...:)` in extension.
+            private init() {}
+        }
+
+        /// Frequency for collecting RUM vitals.
+        public enum VitalsFrequency: String {
+            /// Every `100ms`.
+            case frequent
+            /// Every `500ms`.
+            case average
+            /// Every `1000ms`.
+            case rare
+        }
+
+        // MARK: - Additional Interface For Datadog Cross-Platform SDKs
 
-public typealias URLSessionRUMAttributesProvider = (URLRequest, URLResponse?, Data?, Error?) -> [AttributeKey: AttributeValue]?
-
-public struct RUMConfiguration {
-    public struct Instrumentation {
-        public let uiKitRUMViewsPredicate: UIKitRUMViewsPredicate?
-        public let uiKitRUMUserActionsPredicate: UIKitRUMUserActionsPredicate?
-        public let longTaskThreshold: TimeInterval?
-
-        public init(
-            uiKitRUMViewsPredicate: UIKitRUMViewsPredicate? = nil,
-            uiKitRUMUserActionsPredicate: UIKitRUMUserActionsPredicate? = nil,
-            longTaskThreshold: TimeInterval? = nil
-        ) {
-            self.uiKitRUMViewsPredicate = uiKitRUMViewsPredicate
-            self.uiKitRUMUserActionsPredicate = uiKitRUMUserActionsPredicate
-            self.longTaskThreshold = longTaskThreshold
+        /// Grants access to an internal interface utilized only by Datadog cross-platform SDKs.
+        /// **It is not meant for public use** and it might change without prior notice.
+        public var _internal = InternalConfiguration()
+
+        /// An interface granting access to internal methods exclusively utilized by Datadog cross-platform SDKs.
+        /// **It is not meant for public use.**
+        ///
+        /// Methods, members, and functionality of this interface is subject to change without prior notice,
+        /// as they are not considered part of the public interface of the Datadog SDK.
+        public struct InternalConfiguration {
+            /// The sampling rate for configuration telemetry events. When set, it overwrites the value
+            /// of `configurationTelemetrySampleRate` in `RUM.Configuration`.
+            ///
+            /// It is mostly used to enable or disable telemetry events when running test scenarios.
+            /// Expects value between `0.0` and `100.0`.
+            public var configurationTelemetrySampleRate: Float? = nil
         }
```

Comment:
You don't need this anymore with the addition of `InternalExtended`. You can now do:
```swift
extension RUM {
  struct Configuration {
    /// An extra sampling rate for configuration telemetry events.
    ///
    /// It is applied on top of the value configured in public `telemetrySampleRate`.
    internal var configurationTelemetrySampleRate: Float = 20.0
  }
}

extension RUM.Configuration: InternalExtended {}
extension InternalExtension where ExtendedType == RUM.Configuration {
    /// The sampling rate for configuration telemetry events. When set, it overwrites the value
    /// of `configurationTelemetrySampleRate` in `RUM.Configuration`.
    ///
    /// It is mostly used to enable or disable telemetry events when running test scenarios.
    /// Expects value between `0.0` and `100.0`.
    public var configurationTelemetrySampleRate: Float {
        get { type.configurationTelemetrySampleRate }
        set { type.configurationTelemetrySampleRate = newValue }
    }
}

var config = RUM.Configuration()
config._internal_mutation {
  $0.configurationTelemetrySampleRate = 42
}
```

It will be easier for CP to use the same pattern accross config 👍

Reply:
Excellent 💪, done ✅!